File: enrollment-service/src/main/java/com/cts/edutrack/model/Enrollment.java
package com.cts.edutrack.model;

import java.time.LocalDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Data
public class Enrollment {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long enrollmentId;
	private Long programId;
	private Long userId;
	private LocalDateTime enrolledDate;
	private String status;
}

File: EduTrackFrontEnd/src/app/core/models/enrollment.ts
export interface EnrollmentRequest {
  programId: number;
  userId: number;
}

export interface EnrollmentResponse {
  enrollmentId: number;
  programId: number;
  userId: number;
  enrolledDate: string;
  status: 'Active' | 'Completed' | 'Dropped';
}

export interface EnrollmentStatusUpdateRequest {
  status: string;
}